### PR TITLE
Replace Bazel with Gradle

### LINKED ISSUE #225: Issues with Bazel build
I encountered a few practical issues with Bazel, some of them arising only with its latest version (0.21): a scarce IDEs support (Eclipse wouldn't create a project because of the Bazel .project folder, which prevents Eclipse from creating the .project file) and building failing due to some missing deprecation flags (`--incompatible_package_name_is_a_function=false`, `--incompatible_remove_native_http_archive=false`).  
Please consider switching to a more developer-friendly building tool (related: https://github.com/lupino3/edumips64/issues/180#issuecomment-453864887 ).

### PR DESCRIPTION
**Pull request by pviotti**
This PR replaces Bazel with Gradle. This should ease the maintenance burden (all configurations are in `build.gradle`) and make the project easier to approach for developers.  
More information on Gradle tasks are available in the developer guide, updated in this PR.  

These are the remaining TODOs (AFAIK):

- [ ] build / check scripts for GWT web ui, devmode (?) 
- [ ] minor: remove images and docs from cli jar
- [ ] update gh pages script and uncomment corresponding lines in .travis.yml 

(I suggest squashing upon merge: there's quite a few throwaway commits to fix the travis builds..)

closes #225

### CODE REVIEW

File: docs/developer-guide.md [hunk 1]
```diff
@@ -36,38 +33,30 @@ should have Automatic Style (astyle) installed.
...
+### Main Gradle tasks
 
-Note that you will need to specify the command to use to generate build-time
-variables. This is done by appending the following parameters to all the
-`bazel build` commands: `--workspace_status_command=utils/bazel-stamp.sh --stamp`.
+All the tasks of Gradle 
+[Java](https://docs.gradle.org/current/userguide/java_plugin.html#sec:java_tasks) and 
+[Application](https://docs.gradle.org/current/userguide/application_plugin.html#sec:application_tasks) 
+plugins are available to build,
+compile documentation, run tests and run EduMips64 itself.
```
Comment (lupino3) [inline docs/developer-guide.md]:
```suggestion
compile documentation, run tests and run EduMIPS64 itself.  
```

File: docs/developer-guide.md [hunk 2]
```diff
@@ -36,38 +33,30 @@ should have Automatic Style (astyle) installed.
...
+[Application](https://docs.gradle.org/current/userguide/application_plugin.html#sec:application_tasks) 
+plugins are available to build,
+compile documentation, run tests and run EduMips64 itself.  
+In particular you may find useful these tasks:
 
-* `//src/main/java/org/edumips64:slim-jar` builds the jar package named
-  edumips64-`version`-nodeps.jar, that does not embed the JavaHelp libraries
-  and is oriented towards distribution package creators, that should add a
-  dependency on JavaHelp and appropriately set the classpath in their scripts.
+ * `./gradlew assemble`  - (Java plugin) compile and assemble jar artifacts
```
Comment (lupino3) [inline docs/developer-guide.md]:
```suggestion
 * `./gradlew assemble` - (Java plugin) compile and assemble jar artifacts 
```

File: docs/developer-guide.md [hunk 3]
```diff
@@ -36,38 +33,30 @@ should have Automatic Style (astyle) installed.
...
+Gradle builds the following jar artifacts:
 
-* `//src/main/java/org/edumips64:standalone-jar` builds the jar package named
-  edumips64-`version`.jar, that embeds the JavaHelp libraries, and is oriented
-  towards users downloading the JAR archive from the website (not through
-  package managers).
-  
-* `//src/main/java/org/edumips64:cli-jar` builds a jar package containing an
-  experimental CLI front-end.
+ - `edumips64-1.2.5-standalone.jar`: GUI executable jar including the JavaHelp dependency
```
Comment (lupino3) [inline docs/developer-guide.md]: You probably want to abstract away the version, so we don't need to update this file every time we release a new version :)

File: .travis.yml [hunk 4]
```diff
@@ -1,72 +1,29 @@
----
-sudo: false
```
[the remark this answers is not in the document]
Comment (pviotti) [reply .travis.yml]: >  If you currently specify sudo: false in your .travis.yml, we recommend removing that configuration soon.

https://blog.travis-ci.com/2018-10-04-combining-linux-infrastructures

**Pull request merged**